tests/framework/ChangeLogTest.php

```diff
@@ -0,0 +1,54 @@
+<?php
+/**
+ * @link http://www.yiiframework.com/
+ * @copyright Copyright (c) 2008 Yii Software LLC
+ * @license http://www.yiiframework.com/license/
+ */
+
+namespace yiiunit\framework;
+
+use yiiunit\TestCase;
+
+/**
+ * ChangeLogTest.
+ * @group base
+ */
+class ChangeLogTest extends TestCase
+{
+    public function changeProvider()
+    {
+
+        $lines = explode("\n", file_get_contents(__DIR__ . '/../../framework/CHANGELOG.md'));
+
+        // Don't check last 1500 lines, they are old and often don't obey the standard.
+        $lastIndex = count($lines) - 1500;
+        $result = [];
+        foreach($lines as $i => $line) {
```

Comment:
@SamMousa let me be menace and say that it's better to put space after `foreach` for PSR-2 compability :)